File: jsObject/monsterCard.js
"use strict";
card = require('./card.js');
class MonsterCard extends card.Card {
    constructor(name) {
        super(name);
        this._attack=0;
        this._defense=0;
    }
    //Get functions
    get atk(){
        return this._attack;
    }
    get def(){
        return this._defense;
    }
    //Set functions
    set atk(attack){
        this._attack=attack;
    }
    set def(defense){
        this._defense=defense;
    }
}

exports.MonsterCard = MonsterCard;

File: cppDiamond/src/monsterCard.cpp
#include "card.h"
#include "monsterCard.h"

namespace card{
    MonsterCard::MonsterCard(const std::string & _name, int _atk, int _def) 
        : Card(_name), atk(_atk), def(_def){}
    void MonsterCard::setAtk(int _atk){
        atk=_atk;
    }
    void MonsterCard::setDef(int _def){
        def=_def;
    }
    int MonsterCard::getAtk() const{
        return atk;
    }
    int MonsterCard::getDef() const{
        return def;
    }
    MonsterCard::~MonsterCard(){}
}